go.mod

```diff
@@ -12,12 +12,31 @@ require (
 	github.com/spf13/pflag v1.0.5
 	k8s.io/apimachinery v0.21.2
 	k8s.io/client-go v11.0.0+incompatible
-	k8s.io/klog/v2 v2.8.0
+	k8s.io/klog/v2 v2.60.1
 	sigs.k8s.io/yaml v1.2.0
 )
 
 require (
-	github.com/go-logr/logr v0.4.0 // indirect
+	github.com/Microsoft/go-winio v0.4.16 // indirect
+	github.com/ProtonMail/go-crypto v0.0.0-20210428141323-04723f9f07d7 // indirect
+	github.com/acomagu/bufpipe v1.0.3 // indirect
+	github.com/emirpasic/gods v1.12.0 // indirect
+	github.com/go-git/gcfg v1.5.0 // indirect
+	github.com/go-git/go-billy/v5 v5.3.1 // indirect
+	github.com/go-git/go-git-fixtures/v4 v4.3.1 // indirect
+	github.com/imdario/mergo v0.3.12 // indirect
+	github.com/jbenet/go-context v0.0.0-20150711004518-d14ea06fba99 // indirect
+	github.com/kevinburke/ssh_config v0.0.0-20201106050909-4977a11b4351 // indirect
+	github.com/mitchellh/go-homedir v1.1.0 // indirect
+	github.com/sergi/go-diff v1.1.0 // indirect
+	github.com/xanzy/ssh-agent v0.3.0 // indirect
+	golang.org/x/crypto v0.0.0-20210421170649-83a5a9bb288b // indirect
+	gopkg.in/warnings.v0 v0.1.2 // indirect
+)
+
+require (
+	github.com/go-git/go-git/v5 v5.4.2
```

Comment:
might sound dumb but since `gitutil` pkg itself is implemented using exec(git), could we also just use that for the test helpers? not a must, especially since it's a test-only dependency. I'm curious why you went down this path.

I recall we initially tried to implement `gitutil` with go-git but I don't recall why we didn't pursue that path (it might be that the outputs/err msgs from `git` cli tool are more helpful and recognizable).

Reply:
So now what i thought is, to keep the approach consistent across the repo, i made changes with git cli. 

Thank you very much for the explanation !!